examples/testbed/2d.rs

```diff
@@ -0,0 +1,172 @@
+//! 2d testbed
+//!
+//! You can switch scene by pressing the spacebar
+
+#[cfg(feature = "bevy_ci_testing")]
+use bevy::dev_tools::ci_testing::CiTestingCustomEvent;
+use bevy::prelude::*;
+
+fn main() {
+    let mut app = App::new();
+    app.add_plugins((DefaultPlugins,))
+        .init_state::<Scene>()
+        .enable_state_scoped_entities::<Scene>()
+        .add_systems(OnEnter(Scene::Shapes), shapes::setup)
+        .add_systems(OnEnter(Scene::Bloom), bloom::setup)
+        .add_systems(OnEnter(Scene::Text), text::setup)
+        .add_systems(OnEnter(Scene::Sprite), sprite::setup)
+        .add_systems(Update, switch_scene);
+    app.run();
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, States, Default)]
+enum Scene {
+    #[default]
+    Shapes,
+    Bloom,
+    Text,
+    Sprite,
+}
+
+fn switch_scene(
+    keyboard: Res<ButtonInput<KeyCode>>,
+    #[cfg(feature = "bevy_ci_testing")] mut ci_events: ResMut<Events<CiTestingCustomEvent>>,
+    scene: Res<State<Scene>>,
+    mut next_scene: ResMut<NextState<Scene>>,
+) {
+    let mut should_switch = false;
+    should_switch |= keyboard.just_pressed(KeyCode::Space);
+    #[cfg(feature = "bevy_ci_testing")]
+    {
+        should_switch |= !ci_events.is_empty();
```

Comment:
I would prefer this actually filter for the `switch_scene` command, so it doesn't break when other custom commands are added in the future.